Lib/ufo2ft/filters/propagateAnchors.py

```diff
@@ -76,17 +76,18 @@ def _propagate_glyph_anchors(glyphSet, composite, processed):
                 base_components.append(component)
                 anchor_names |= {a.name for a in glyph.anchors}
 
-    if mark_components and not base_components and _is_ligature_mark(composite):
-        # The composite is a mark that is composed of other marks (E.g.
-        # "circumflexcomb_tildecomb"). Promote the mark that is positioned closest
-        # to the origin to a base.
-        try:
-            component = _component_closest_to_origin(mark_components, glyphSet)
-        except Exception as e:
-            raise Exception(
-                "Error while determining which component of composite "
-                "'{}' is the lowest: {}".format(composite.name, str(e))
-            )
+    if mark_components and not base_components:
+        if _is_ligature_mark(composite):
+            # The composite is a mark that is composed of other marks (E.g.
+            # "circumflexcomb_tildecomb"). Promote the mark that is positioned
+            # closest to the origin to a base.
+            try:
+                component = _component_closest_to_origin(mark_components, glyphSet)
+            except Exception as e:
+                raise Exception(
+                    "Error while determining which component of composite "
+                    "'{}' is the lowest: {}".format(composite.name, str(e))
+                )
         mark_components.remove(component)
```

Comment:
where is the local variable `component` being defined/assigned? I only see an explicit assignment inside the `if _is_ligature_mark(composite)` block. What if the if statement is false?

Reply:
Probably needs a `else: component = composite`?

Reply:
Ouch I should have seen that. Fixed in new commits.